process/block/metablock.go

```diff
@@ -291,22 +290,22 @@ func (mp *metaProcessor) removeBlockInfoFromPool(header *block.MetaBlock) error
 		return process.ErrNilHeadersNoncesDataPool
 	}
 
-	for i := 0; i < len(header.ShardInfo); i++ {
-		shardData := header.ShardInfo[i]
-
-		obj, ok := headerPool.Peek(shardData.HeaderHash)
-		if !ok {
+	mp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
+	for shardBlockHash, hdrInfo := range mp.hdrsForCurrBlock.hdrHashAndInfo {
```

Comment:
it is much cleaner when you iterate on a set header, like the old code. You still could do the same as the old code, and instead of peeking from data pool you get the data from the hdrsForCurrBlock

Reply:
Yes, I think about this but I did it in this way in my both implementation (EN-4211 and this one) as I reduced the number of parameters and also some code related to getting values from this new map and checking them for nil, if I would used your proposal. We can refactor all them back, just because I agree that is more cleaner in this way, in the new task which I have added in Jira EN-4278 for refactor the management of requested/received of missed: txs, shard blocks, meta blocks into separate component.

Reply:
we might forget that this refactoring is needed in that Jira task - so add the requirements there, or the comments from this pull request. It is not necessarily good to have all these change requests postponed for another PR, this is an easy change which could be done for this PR.

Reply:
Done in this PR